### ISSUE TITLE
Question: function_parameter_count rule regarding initializers

### ISSUE THREAD

**Issue by mcplectrum** [realm/SwiftLint#653]
Hi,

we have a business model class (here: **Example**), which looks like the following:

```swift
class Example {
  let a: String
  let b: String
  let c: String
  let d: String
  let e: String
  let f: String

  init(a: String, b: String, c: String, d: String, e: String, f: String) {
    self.a = a
    self.b = b
    ...
 }
}
```

The initializer is used after parsing the values from a JSON.

Without configuring the rule: **function_parameter_count**,  it would show a warning here, because there are > 5 params.

So my question is, how to solve this issue, without changing the config:
- Should this rule be modified and ignore init params?
- Is the model itself wrong, because it has too many properties?
- Should i go around this warning by artificially grouping properties that do not have anything in common?
- Should i substitute this initializer with one that takes the JSON object as a single parameter? <= this would move the logic of the JSON structure inside the class, which should be avoided.
- Does someone has a better approach / idea here?

**Comment by jpsim** [comment]
This is a duplicate of #544.
